client/common/cmdline.c

```diff
@@ -1848,28 +1849,25 @@ int freerdp_client_settings_parse_command_line_arguments(rdpSettings* settings,
 		}
 		CommandLineSwitchCase(arg, "kbd")
 		{
-			unsigned long id;
-			char* pEnd;
-			id = strtoul(arg->Value, &pEnd, 16);
-
-			if (pEnd != (arg->Value + strlen(arg->Value)))
-				id = 0;
+			unsigned long id = strtoul(arg->Value, NULL, 0);
 
-			if (id == 0)
+			if ((errno != 0) || (id > UINT32_MAX) || (id == 0))
 			{
 				const int rc = freerdp_map_keyboard_layout_name_to_id(arg->Value);
 
 				if (rc < 0)
```

Comment:
This check is unnecessary now that we check for ```<= 0``` 2 lines below

Reply:
@mfleisz You're in the wrong scope..

Reply:
@akallabeth I don't see how I'm in the wrong scope ... there is a check for < 0 at line 1858 and another for <= 0 at line 1861 - both within the scope of ```if ((errno != 0) || (id > UINT32_MAX) || (id == 0))```

Reply:
@mfleisz Ah, that is what you mean, thought you were referring to line 1854 
Yes, the check is to add an additional error message. But you're right, that one is not really necessary.